tests/Tests.elm

```diff
@@ -73,111 +73,173 @@ expectationTests =
+                -- in a 64bit float, so we're just going to ignore that one.
+                float64epsilon =
+                    2.0 ^ -52
 
-                        delta =
-                            abs <| toFloat a
-                    in
-                        da |> Expect.within delta -da
-            , test "Plus minus minNormal equality" <|
-                \() ->
-                    let
-                        float64minNormal =
-                            2 ^ -1022
-                    in
-                        float64minNormal |> Expect.within float64minNormal float64minNormal
-            , test "Very large float equality" <|
-                \() ->
-                    let
-                        float64maxValue =
-                            (2 - (2 ^ -52)) * 2 ^ 1023
-                    in
-                        float64maxValue |> Expect.within 1 float64maxValue
-            , test "Very small float equality" <|
-                \() -> 2 ^ -1022 |> Expect.within 1 (2 ^ -1022)
-            , test "Very small plus minus float equality" <|
-                \() -> 2 ^ -1022 |> Expect.within 1 (2 ^ -1022)
-            , test "Very large difference float equality" <|
-                \() -> 2 ^ 1022 |> Expect.notWithin 1 (-(2 ^ 1022))
-            , fuzz4 float float float float "Within = not notWithin" <|
-                \epsilon a b delta ->
-                    let
-                        isWithin =
-                            Expect.within delta a b
+                infinity =
+                    1.0 / 0.0
 
-                        isNotWithin =
-                            Expect.notWithin delta a b
-                    in
-                        Expect.notEqual (succeeded isWithin) (succeeded isNotWithin)
-            , fuzz3 float float float "within commutativity" <|
-                \epsilon a b ->
-                    succeeded (Expect.within epsilon a b) |> Expect.equal (succeeded <| Expect.within epsilon b a)
-            , fuzz3 float float float "notWithin commutativity" <|
-                \epsilon a b ->
-                    succeeded (Expect.notWithin epsilon a b) |> Expect.equal (succeeded <| Expect.notWithin epsilon b a)
-            , fuzz2 float float "within reflexive" <|
-                \epsilon a ->
-                    Expect.within epsilon a a
-            , expectToFail <|
-                fuzz2 float float "notWithin reflexive" <|
+                -- Not a Number
+                nan =
+                    0.0 / 0.0
+
+                -- Smallest possible value before we start losing precision
+                float64minNormal =
+                    2.0 ^ -1022
+
+                -- Largest finite value representable in a 64bit float
+                float64maxValue =
+                    (2.0 - float64epsilon) * 2.0 ^ 1023
+
+                -- Smallest finite positive value representable in a 64bit float
+                float64minValue =
+                    2.0 ^ -1074
```

Comment:
These constants are duplicated. Are they available in some core module? Should they be in a `Floats.elm` file in this package?